src/shaders/fill_extrusion.fragment.glsl

```diff
@@ -1,7 +1,23 @@
+
+uniform lowp vec3 u_lightpos;
+
 varying vec4 v_color;
 
+#ifdef RENDER_SHADOWS
+varying vec4 v_pos_light_view_0;
+varying vec4 v_pos_light_view_1;
+varying float v_depth;
+varying vec3 v_normal;
+#endif
+
 void main() {
     vec4 color = v_color;
+
+#ifdef RENDER_SHADOWS
+    color.xyz = shadowed_color_normal(color.xyz, normalize(v_normal), normalize(u_lightpos),
```

Comment:
Could `u_lightpos` be normalized on the CPU instead?

Reply:
The vector is used unnormalized in the vertex shader.

Reply:
Did you check whether the `u_lightpos` is already normalized before passing to the shaders? Usage of the variable in the vertex shader would indicate that it's already normalized.

Reply:
It is not normalized on the CPU. The radial coordinate (distance) is part of the light specification and I believe this is the reason why it is used unnormalized.